controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -2718,5 +2723,21 @@ private static boolean isOperator(HttpRequest request) {
                               .anyMatch(definition -> definition == RoleDefinition.hostedOperator);
     }
 
+    private void ensureApplicationExistsForPublic(TenantAndApplicationId id, HttpRequest request) {
+        if (controller.system().isPublic() && controller.applications().getApplication(id).isEmpty()) {
+            log.fine("Application does not exist in public, creating: " + id);
+            var credentials = accessControlRequests.credentials(id.tenant(), null /* not used on public */ , request.getJDiscRequest());
+            controller.applications().createApplication(id, credentials);
+        }
+    }
+
+    private void ensureInstanceExistsForPublic(ApplicationId id, HttpRequest request) {
```

Comment:
Not necessary, instance is created if it doesn't exist in `JobController::deploy`. Verified this already works by deploying to a new application without any instances.

Reply:
Alright, will remove.